contracts/mocks/MockMultiModule.sol

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+import { IModule } from "contracts/interfaces/modules/IModule.sol";
+import { EncodedModuleTypes } from "contracts/lib/ModuleTypeLib.sol";
+import { PackedUserOperation } from "account-abstraction/contracts/interfaces/PackedUserOperation.sol";
+import { ECDSA } from "solady/src/utils/ECDSA.sol";
+import { SignatureCheckerLib } from "solady/src/utils/SignatureCheckerLib.sol";
+import { MessageHashUtils } from "@openzeppelin/contracts/utils/cryptography/MessageHashUtils.sol";
+import "contracts/types/Constants.sol";
+
+contract MockMultiModule is IModule {
+
+    mapping(uint256 moduleTypeId => mapping (address smartAccount => bytes32 initData)) configs;
+
+    function validateUserOp(PackedUserOperation calldata userOp, bytes32 userOpHash) external view returns (uint256 validation) {
+        address owner = address(bytes20(configs[MODULE_TYPE_VALIDATOR][msg.sender]));
+        return
+            ECDSA.recover(MessageHashUtils.toEthSignedMessageHash(userOpHash), userOp.signature) == owner
+                ? VALIDATION_SUCCESS
+                : VALIDATION_FAILED;
+    }
+
+    function getConfig(address smartAccount, uint256 moduleTypeId) external view returns (bytes32) {
+        return configs[moduleTypeId][smartAccount];
+    }
+
+    function onInstall(bytes calldata data) external override {
+        if (data.length >= 0x21) {
+            uint256 moduleTypeId = uint256(uint8(bytes1(data[:1])));
+            configs[moduleTypeId][msg.sender] = bytes32(data[1:33]);
```

Comment:
didn't get this logic

Reply:
just storing the config data based on module type. it is just a mock though